### ISSUE TITLE
Why this sample is not using Camera2 API?

### ISSUE THREAD

**Issue by bhavesh-hirpara** [firebase/quickstart-android#677]
Will it be possible with camera 2 API?

**Comment by samtstern** [comment]
cc @gkaldev

**Comment by jeduden** [comment]
@the-dagger  the [android documentation says the following](https://developer.android.com/reference/android/graphics/ImageFormat#YV12):
> For the android.hardware.camera2 API, the YUV_420_888 format is recommended for YUV output instead.

While only NV21 and YV21 are supported by the [firebase ml sdk](https://firebase.google.com/docs/reference/android/com/google/firebase/ml/vision/common/FirebaseVisionImageMetadata) ? 
are we supposed convert on the fly wasting cpu cycles/battery or just configure camera2 to out NV21 ? 
but then going against the recommendation ? 
also i am not clear what drawbacks the recommendation tries to avoid. 

The [camera2 summary](https://developer.android.com/reference/android/hardware/camera2/package-summary) also repeats this recommendation.

**Comment by the-dagger** [comment]
I think @samtstern or @gkaldev might be the best people to answer this.

**Comment by mr-gil** [comment]
why this example is not migrated to camera2?

**Comment by GuanacoDevs** [comment]
If you don't need any of the fancy stuff that comes with Camera2, I highly recommend to use [CameraView](https://github.com/natario1/CameraView).
I made my custom Camera2 to process frames, based from this sample, however this sample uses Texture View, when seems like Surface Texture might be more appropriate.
I did ship with my custom Camera2 and [CameraView](https://github.com/natario1/CameraView)(for legacy devices) and [CameraView](https://github.com/natario1/CameraView) is waaaaay more efficient than my own Camera2.
Best part of [CameraView](https://github.com/natario1/CameraView) is that I only wrote less than 10 lines of java code to have a very efficient camera for OCR Reader along with a QR Reader.
